pkg/cmd/run-pgo-build/main.go

```diff
@@ -0,0 +1,268 @@
+		return nil, 0, err
+	}
+	_, err = zipFile.Seek(0, 0)
+	if err != nil {
+		zipFile.Close()
+		return nil, 0, err
+	}
+
+	return zipFile, zipLength, nil
+}
+
+// processArtifactsZip processes a file named artifacts.zip inside the larger
+// artifacts zip archive. This is confusing, but the `.pprof` files for any
+// given node are inside a sub-zip archive named `artifacts.zip`, which is one
+// of the files in the giant zip archive that we fetch for the build. This
+// function processes that sub-zip archive and extracts all the CPU (.pprof)
+// files.
+//
+// The .pprof files will be parsed and put into the pprofFilesChan.
+// wg.Done() will be called when this function completes.
+func processArtifactsZip(f *zip.File, profilesChan chan *profile.Profile, wg *sync.WaitGroup) {
+	archive, err := f.Open()
+	if err != nil {
+		panic(err)
+	}
+	defer archive.Close()
+	var archiveBuf bytes.Buffer
+	_, err = io.Copy(&archiveBuf, archive)
+	zipReader, err := zip.NewReader(bytes.NewReader(archiveBuf.Bytes()), int64(f.UncompressedSize64))
+	if err != nil {
+		panic(err)
+	}
+	for _, file := range zipReader.File {
+		if strings.HasSuffix(file.FileHeader.Name, ".pprof") &&
+			strings.Contains(file.FileHeader.Name, "/cpuprof.") &&
+			file.UncompressedSize64 > 0 {
+			pprofFile, err := file.Open()
+			if err != nil {
+				panic(err)
+			}
+			defer pprofFile.Close()
+			prof, err := profile.Parse(pprofFile)
+			if err != nil {
+				panic(err)
+			}
+			profilesChan <- prof
+		}
+	}
+	wg.Done()
+}
+
+// processPprofFiles reads all the parsed profiles from the given channel,
+// merges all of the profiles, and dumps the results profile to a final
+// location. wg.Done() will be called at the end of this function.
+func processPprofFiles(profilesChan chan *profile.Profile, wg *sync.WaitGroup) {
+	var profiles []*profile.Profile
+	for prof := range profilesChan {
+		profiles = append(profiles, prof)
+	}
+	if len(profiles) == 0 {
+		panic("expected to find a profile; found none")
+	}
+	res, err := profile.Merge(profiles)
+	if err != nil {
+		panic(err)
+	}
+	w, err := os.Create(*outFile)
+	if err != nil {
+		panic(err)
+	}
+	defer w.Close()
+	err = res.Write(w)
+	if err != nil {
+		panic(err)
+	}
+	wg.Done()
+}
+
+func main() {
+	flag.Parse()
+
+	if branch == "" || serverURL == "" || username == "" || password == "" {
+		panic("ensure credentials, server URL, and branch are supplied")
+	}
+	if *outFile == "" {
+		panic("must supply -out")
+	}
+
+	tmpDir, err := os.MkdirTemp("", "run-pgo-build")
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		_ = os.RemoveAll(tmpDir)
+	}()
+
+	build, err := queueBuild(buildConfigID, branch)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("queued build with ID %d\n", build.ID)
+
+	for {
+		if build.FinishDate != "" {
+			break
+		}
+		time.Sleep(3 * time.Minute)
+		lookupID := build.ID
+		newBuild, err := getBuild(lookupID)
+		if err != nil {
+			fmt.Printf("failed to get build %d; got error %+v\n", lookupID, err)
```

Comment:
A nit. Can you add some wording that this is expected and the operation will be retried.